src/yunohost/settings.py

```diff
@@ -0,0 +1,189 @@
+import os
+import json
+import errno
+
+from datetime import datetime
+from collections import OrderedDict
+
+from moulinette.core import MoulinetteError
+from moulinette.utils.log import getActionLogger
+
+logger = getActionLogger('yunohost.settings')
+
+SETTINGS_PATH = "/etc/yunohost/settings.json"
+SETTINGS_PATH_OTHER_LOCATION = "/etc/yunohost/settings-%s.json"
+
+# a settings entry is in the form of:
+# name: {type, value, default, description, [possibilities]}
+# possibilities is only for enum
+
+# description is implied from the translated strings
+# the key is "global_settings_setting_%s" % key.replace(".", "_")
+
+# type can be:
+# * bool
+# * int
+# * string
+# * enum (in form a python list)
+
+# we don't store the value in default options
+DEFAULTS = OrderedDict([
+    ("example.bool", {"type": "bool", "default": True}),
+    ("example.int", {"type": "int", "default": 42}),
+    ("example.string", {"type": "string", "default": "yolo swag"}),
+    ("example.enum", {"type": "enum", "default": "a", "choices": ["a", "b", "c"]}),
```

Comment:
Could we define a list of real-life settings we'd like to have ? Imho that would help to have a better feeling about namespaces and how we'll use it in the future. There could be some things such as : 
```
notifications.mail -> "kload@yoloswag.com"
security.allow_ssh_users_by_default -> "True"
security.fail2ban_harshness ->  [ "low", "mid", "high" ]
security.ssl_security -> [ "low", "high" ] # Would impact the key size for certificates and DH parameters
```